storage/client/client.go

```diff
@@ -1425,13 +1437,59 @@ func (c *StorageClient) RuntimeEvents(ctx context.Context, p apiTypes.GetRuntime
 	return &es, nil
 }
 
+func (c *StorageClient) fetchAccountBalanceFromNode(ctx context.Context, ch chan *RuntimeSdkBalance, runtime common.Runtime, address staking.Address) {
+	runtimeApi, ok := c.runtimeClients[runtime]
+	if !ok {
+		c.logger.Warn("no runtime api configured to fetch account balances", "runtime", runtime)
+		close(ch)
+		return
+	}
+	balances, err := runtimeApi.Client.GetBalances(ctx, roothash.RoundLatest, address)
```

Comment:
I still think we shouldn't fetch at RoundLatest. You explained the motivation for it in the PR description nicely, but most of those arguments only apply when nexus is reindexing, and I don't think we should care much about API robustness or responsiveness during that time.

My primary concern is that by fetching RoundLatest, the DB balance will quite frequently lag behind the node-fetched balance (because nexus has to first notice that a new block arrived, then fully analyze it, then write it to DB). So we'll log some non-negligible amount of warnings about mismatched balances between nexus and node, and we'll never know if it's a genuine mismatch or if the two data sources were just out of sync. That makes logs much much less actionable.

Having good logs here is one of the two main motivations (or maybe even _the_ main motivation) for this PR, i.e. querying the node in nexus and not on the frontend. The other one is that nexus will have less incorrect data, which is nice for non-Explorer users.

Another small benefit of not querying at RoundLatest is that the output will be internally consistent. (Mostly. I guess we should ideally be doing all the DB queries inside a DB tx.)

Reply:
Hm I'd be okay with fetching at the latest indexed round if the frontend (explorer) will continue to fetch the latest balances from the node and overwrite any stale data Nexus might return. Stale balances would likely cause end users to panic, and our sapphire/emerald nodes have had less than perfect uptime. I assumed that the main goal of this PR was to have the explorer stop querying the node on their end though. 

Maybe we could query at the latest indexed round for now as an experiment to find any stale balances in the logs, and once those are fixed we'll start querying at `RoundLatest`?

Reply:
Discussed further in product sync today, and we decided to query for the balance at the current indexed height (not RoundLatest). If Nexus is behind/reindexing, the explorer will warn users appropriately.